battle-service/src/main/java/org/astashonok/battleservice/entities/Battle.java

```diff
@@ -0,0 +1,89 @@
+package org.astashonok.battleservice.entities;
+
+import com.vladmihalcea.hibernate.type.array.DoubleArrayType;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.astashonok.battleservice.constants.BattleConstants;
+import org.astashonok.battleservice.models.BattleStatus;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+
+import javax.persistence.*;
+import java.util.Arrays;
+import java.util.Objects;
+
+@Entity
+@Table(name = "battles")
+@TypeDef(name = "board_array", typeClass = DoubleArrayType.class)
+@Getter
+@Setter
+@NoArgsConstructor
+public class Battle {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private int boardHeight;
+    private int boardWidth;
+
+    @Type(type = "board_array", parameters = @org.hibernate.annotations.Parameter(name = "sql_array_type", value = "integer"))
+    @Column(columnDefinition = "integer[][]")
+    private int[][] board;
+
+    private int winningNumberInRow;
+
+    @Enumerated(EnumType.STRING)
+    private BattleStatus status;
+    private Long xParticipantId;
+    private Long oParticipantId;
+    private Long nextMoveParticipantId;
+    private int remainingFreeMoveCount;
+
+    private Long winnerId;
+
+    public Battle(Long xParticipantId) {
+        this(xParticipantId, BattleConstants.GAME_BOARD_HEIGHT_DEFAULT, BattleConstants.GAME_BOARD_WIDTH_DEFAULT,
+                BattleConstants.WINNING_NUMBER_IN_ROW_DEFAULT);
+    }
+
+    public Battle(Long xParticipantId, int boardHeight, int boardWidth, int winningNumberInRow) {
```

Comment:
you should refactor this to use creational pattern. try to utilize Builder or Factory method instead of having multiple constructors + some method. 
it's really easy for consumer of your interface to mess attributes with the same type
int boardHeight, int boardWidth, int winningNumberInRow

also it's a bad practice to execute logic in constructor (i.e.) initBoard(..)

Reply:
xParticipantId could be final